crates/sp-executor/src/lib.rs

```diff
@@ -132,6 +132,42 @@ impl<Extrinsic: sp_runtime::traits::Extrinsic + Encode> From<Bundle<Extrinsic>>
     }
 }
 
+/// Signed version of [`OpaqueBundle`].
+#[derive(Decode, Encode, TypeInfo, PartialEq, Eq, Clone, RuntimeDebug)]
+pub struct SignedOpaqueBundle {
+    /// The bundle header.
+    pub opaque_bundle: OpaqueBundle,
+    /// Signature of the opaque bundle.
+    pub signature: ExecutorSignature,
+    /// Signer of the signature.
+    pub signer: ExecutorId,
+}
+
+impl SignedOpaqueBundle {
+    /// Returns the hash of inner opaque bundle.
+    pub fn hash(&self) -> H256 {
+        self.opaque_bundle.hash()
+    }
+}
+
+impl<Extrinsic: sp_runtime::traits::Extrinsic + Encode> From<SignedBundle<Extrinsic>>
```

Comment:
`sp_runtime::traits::Extrinsic` do we need the trait bound?

Reply:
Good catch, it's now unnecessary.